src/Mod/Path/PathScripts/PathDrilling.py

```diff
@@ -171,23 +172,34 @@ def circularHoleExecute(self, obj, holes):
             self.commandlist.append(Path.Command('G0', {'Z': obj.SafeHeight.Value, 'F': self.vertRapid}))
             self.commandlist.append(Path.Command('G0', {lastAxis: 0.0, 'F': self.axialRapid}))
 
-    def opSetDefaultValues(self, obj, job):
-        '''opSetDefaultValues(obj, job) ... set default value for RetractHeight'''
+    def opSetDefaultRetractHeight(self, obj, job=None):
+        '''opSetDefaultRetractHeight(obj, job) ... set default Retract Height value'''
 
-        parentJob = PathUtils.findParentJob(obj)
+        has_job = True
+        if not job:
+            job = PathUtils.findParentJob(obj)
+            has_job = False
 
-        if hasattr(parentJob.SetupSheet, 'RetractHeight'):
-            obj.RetractHeight = parentJob.SetupSheet.RetractHeight
+        if hasattr(job.SetupSheet, 'RetractHeight'):
+            obj.RetractHeight = job.SetupSheet.RetractHeight
         elif self.applyExpression(obj, 'RetractHeight', 'OpStartDepth+1mm'):
-            obj.RetractHeight = 10
+            if has_job:
+                obj.RetractHeight = 10
+            else:
+                obj.RetractHeight.Value = obj.StartDepth.Value + 1.0
```

Comment:
I don't understand what's happening here. why is there a difference in behaviour if the job is passed in or if the job is determined?

Reply:
If no `job` is passed in, then it is a method call due to rotation usage and needs to update the RetractHeight because the StartDepth default will be adjusted due to rotation.  Otherwise, the method is being called during op setup when `job` should be passed and is purposed to set the initial default value for RetractHeight.

Pulling out the rotational code will simplify PathWB code complexity and readability again.